### DISCUSSION TITLE
insufficient permissions

### DISCUSSION THREAD

**Discussion by Fred-VIE** [liquidctl/liquidctl#453] [Q&A]
Hello, I am new to Linux (atm on Mint) and tried to install liquidctl.

When I am trying to initialize it says "ERROR: insufficient permissions to access Corsair Hydro H115i Pro".

After installing liquidctl I found that the modules-load.conf was not installed so I created it and put it into /etc/modules-load.d/ as to the instructions and logged out/in.

What am I missing?

**Comment by jonasmalacofilho** [comment]
Hi Fred,

> When I am trying to initialize it says "ERROR: insufficient permissions to access Corsair Hydro H115i Pro".

https://github.com/liquidctl/liquidctl#access-denied-or-open-failed-linux

> After installing liquidctl I found that the modules-load.conf was not installed so I created it and put it into /etc/modules-load.d/ as to the instructions and logged out/in.

We don't use `modules-load.conf` anymore... which version are you running?

Anyway, kernel modules aren't directly related to these permission issues.  What you want is `71-liquidctl.rules`.

**Comment by Fred-VIE** [comment]
I have installed the udev rules in /usr/lib/udev/rules.d/ before I tried to initialize.

Also I tried it under /etc/udev/rules.d/ but the initialization still fails.

The version is 1.9.1

The command I was using to initialize is "liquidctl --match corsair initialize" because I dont want to initialize all devices.

I used the modules-load.conf because the initialization kept failing but it is now deleted again.

**Comment by Fred-VIE** [comment]
Is it normal that this command prompts a "command not found" error ?
sudo liquidctl --match corsair initialize

it does not do that without sudo - still trying to figure out why it wont initialize

**Comment by jonasmalacofilho** [comment]
> I have installed the udev rules in /usr/lib/udev/rules.d/ before I tried to initialize.
> 
> Also I tried it under /etc/udev/rules.d/ but the initialization still fails.

Just to rule out other things: have you restarted since the rules were put in place?

Also, when you installed the rules, did you follow the instructions to reload and re-trigger them (before testing)?

> Is it normal that this command prompts a "command not found" error ?

No, this means that your `root` user has a different `PATH` and the location you installed liquidctl into isn't in it.

**Reply by Fred-VIE** [reply]
> Just to rule out other things: have you restarted since the rules were put in place?

plenty 

> Also, when you installed the rules, did you follow the instructions to reload and re-trigger them (before testing)?

Is that this part? https://github.com/liquidctl/liquidctl#set-up-linux-using-systemd

> No, this means that your `root` user has a different `PATH` and the location you installed liquidctl into isn't in it.

Sorry I dont know what that means. We are just root users for a specific path?
How can I be root user for another path when that is needed?

I did not choose a location for installation. liquidctl did when I used the pip command.

**Reply by jonasmalacofilho** [reply]
> Is that this part? https://github.com/liquidctl/liquidctl#set-up-linux-using-systemd

No, [these](https://github.com/liquidctl/liquidctl/blob/71aca8703c0326a492f41191d504006c6aaf1564/extra/linux/71-liquidctl.rules#L18-L26).  But once you restarted, you should have been good either way.

(Are you by any chance on a very old Mint version, or using any sort of non-standard systemd configuration?)

> Sorry I dont know what that means. We are just root users for a specific path?
> How can I be root user for another path when that is needed?

No, I was referring to the [`PATH` environment variable](https://en.wikipedia.org/wiki/PATH_(variable)), which tells where executables are searched for.  This can (and usually does) vary per user.  There is a way (that I always forget, but you can probably find it online quite easily) to set up your system so that `sudo` does indeed _inherit_ your user's environment, including its `PATH`.

> I did not choose a location for installation. liquidctl did when I used the pip command.

It's possible that the pip command your normal user has access to installed it in a location `root` doesn't look in.  Try running `which liquidctl` and `sudo which liquidctl` and compare the results.

Also, _if the locations are different,_ you could have different liquidctl versions in each: run `<full path to liquidctl> --version` for each location and compare.

**Reply by Fred-VIE** [reply]
> No, [these](https://github.com/liquidctl/liquidctl/blob/71aca8703c0326a492f41191d504006c6aaf1564/extra/linux/71-liquidctl.rules#L18-L26). But once you restarted, you should have been good either way.

I copied in the rule and restarted 

> (Are you by any chance on a very old Mint version, or using any sort of non-standard systemd configuration?)

V20.3 - not sure what a standard or non standard systemd config would be, sorry

>  There is a way (that I always forget, but you can probably find it online quite easily) to set up your system so that `sudo` does indeed _inherit_ your user's environment, including its `PATH`.

That would be useful
 
> Try running `which liquidctl` and `sudo which liquidctl` and compare the results.

With sudo I dont get a path shown :/  I guess thats an indicator that something aint right?
 
> Also, _if the locations are different,_ you could have different liquidctl versions in each: run `<full path to liquidctl> --version` for each location and compare.

at the home path its 1.9.1 and with sudo I get no path 

> No, I was referring to the [`PATH` environment variable](https://en.wikipedia.org/wiki/PATH_(variable)), > 

Reading this is like reading chinese to me.

**Comment by Fred-VIE** [comment]
Could that be the reason for my issue? 

I have installed Kubuntu and installed liquidctl with the pip command there. On installation it says : "WARNING: The script liquidctl is installed in '/home/profilename/.local/bin' which is not on PATH.   Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location."

Well I have checked and /home/profilename/.local/bin is actually on PATH so I have no clue why I got the message that it aint.

I hazily remember it gave the same Warning in Mint as well.

After copying the rules and trying to use the command udevadm control --reload I got the error: 
"Failed to send reload request: Permission denied"

When using the reload and trigger commands with sudo it says nothing and also does not seem to do anything because initialization gives the same error like with Mint.

**Comment by Fred-VIE** [comment]
Since you are not willing to help with problems you should take liquidctl off of github. I have wasted over over half a month on this. It is not like we do not have anything else to do with out lifes.

**Reply by jonasmalacofilho** [reply]
You said it best: It is not like we do not have anything else to do with out lifes.

Locked and blocked.
